### PR TITLE
Fix for the Recycler template.  Use rm to scrub volume, include . pre…

### PR DESCRIPTION
**Pull request by childsb**
This change fixes the recycler template CLI.  The existing template fails with error "find: unrecognized: -delete": https://github.com/kubernetes/kubernetes/pull/16723#issuecomment-177784069

### CODE REVIEW

File: pkg/volume/plugins.go [hunk 1]
```diff
@@ -416,7 +416,7 @@ func NewPersistentVolumeRecyclerPodTemplate() *api.Pod {
 					Name:    "pv-recycler",
 					Image:   "gcr.io/google_containers/busybox",
 					Command: []string{"/bin/sh"},
-					Args:    []string{"-c", "test -e /scrub && echo $(date) > /scrub/trash.txt && find /scrub -mindepth 1 -maxdepth 1 -delete && test -z \"$(ls -A /scrub)\" || exit 1"},
+					Args:    []string{"-c", "test -e /scrub && rm -rf /scrub/..?* /scrub/.[!.]* /scrub/* ; echo $(date) > /scrub/trash.txt && exit 0 "},
```
Comment (markturansky) [inline pkg/volume/plugins.go]: In the original bash, writing the date to 'trash' happened first to ensure something the volume be written to, not empty, and then verified empty afterwards.  You moved it to after the 'rm' command?  I don't see the confirmation that the dir is empty (checking via 'ls')

**Comment by markturansky** [comment]
@kubernetes/rh-storage anyone with Bash expertise?

File: pkg/volume/plugins.go (hunk 1 above)
Comment (jsafrane) [inline pkg/volume/plugins.go]: I confirm Mark's finding. I think this is correct:
```
rm -rf /scrub/..?* /scrub/.[!.]* /scrub/*  && test -z \"$(ls -A /scrub)\" || exit 1
```
(I.e. remove trash.txt, we don't need it and add check)

Comment (markturansky) [inline pkg/volume/plugins.go]: Agreed we don't need trash.

Comment (childsb) [inline pkg/volume/plugins.go]: I thought the "trash" was a status object -- IE notify the user they are using a recycled volume.  If its not needed @jsafrane line will work.. i'll test and update the PR.

**Comment by parikhyash** [comment]
@markturansky @childsb Any ETA on when this will be merged?

The "find: unrecognized: -delete" bug is completely blocking me from using NFS mounts.

**Comment by markturansky** [comment]
@saad-ali add this to the list of PRs for 1.2?  It's a small fix but necessary bug fix.

**Comment by tanen01** [comment]
Would be great if this can be merged asap. This little defect is blocking us from upgrading to k8s v1.1.
Thanks.

**Comment by childsb** [comment]
@markturansky validated and updated the PR with the command discussed above.

@tanen01 we're trying to get this merged asap for 1.2.  As an aside, whats your use cases for NFS-Recycler right now?  We've discussed re-engineering the behavior (https://github.com/kubernetes/kubernetes/pull/19827) and I'd like to understand any current use cases.  Could you chime in on 19827 with your current usage and any enhancements or gaps you currently experience with recycle?

**Comment by saad-ali** [comment]
> @saad-ali add this to the list of PRs for 1.2? It's a small fix but necessary bug fix.

@markturansky once you've LGTM'd it, we can ping on-call for a manual merge to ensure this gets in for 1.2.

**Comment by markturansky** [comment]
LGTM.

**Comment by tanen01** [comment]
@childsb Our use case is a simple one. Each namespace is an "app", and when the app is removed (hence its pvc) we simply want the pv it was bound to be recycled and made available for other apps. We only use GA releases of kubernetes so we really need this bug fix backported to the 1.1 branch. Thank you.

**Comment by saad-ali** [comment]
Merge queue is currently blocked by a failing test suite, but this is 2nd in the queue, once it restarts.

**Comment by saad-ali** [comment]
@k8s-bot test this #IGNORE

**Comment by saad-ali** [comment]
@k8s-bot test this issue: #IGNORE

**Pull request merged**